### PR TITLE
improve clicks emulation by using user-event lib

### PR DESCRIPTION
**Pull request by wKich**
## Motivation

The `element.click()` which is used in HTML interactors doesn't emulate user's clicks well. For that reason, clicks on material-ui buttons don't have a nice ripple effect. Another point is, in some cases elements might have `mousedown` and `mouseup` event listeners and the `element.click()` doesn't emit these events, that leads to different behavior for a user and in tests

## Approach

I think we have to improve actions emulation to real user interaction as close as possible. Instead of writing our own emulation implementation, I thought it's better to use an existing solution like `@testing-library/user-event`. The long-term plan is to improve other actions.

### CODE REVIEW

**Comment by cowboyd** [comment]
Couple of questions:

1. Why are we using `fireEvent` in some cases and `userEvent` in others for a click?
2. I'm not sure how I feel about exporting all of the html symbols.

**Comment by wKich** [comment]
1. The userEvent emulate click with mouseDown and mouseUp events, also it returns void, but for some cases we need to know if click was successful, so there we use fireEvent.
2. Yeah, I also doubt in this solution. But I'd like to reuse testing library in material-ui interactors, the other way is install the same version.

**Comment by cowboyd** [comment]
Is it worth trying to make a PR upstream to make userEvent.click() support returning a value?

Maybe just have a separate entry point called `private` or `testing-library` that we will not support?

**Comment by wKich** [comment]
- I'll take a look. 
- That's a good point

**Comment by wKich** [comment]
@cowboyd about the second point, it should work, but typescript still doesn't have support `exports` feature in package.json https://github.com/microsoft/TypeScript/issues/33079. I'll make some workaround for it.

**Comment by wKich** [comment]
Ugh... It was not so simple to make `testing-library` available in not main entry point. Also check the `userEvent.click` instead of `fireEvent` there is no need to check the cancelation flag, it's already implemented in `testing-library`, so just use it everywhere.

File: packages/html/src/index.ts [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
+export { Interaction, ReadonlyInteraction, isInteraction } from "./interaction";
+export { createInteractor } from "./create-interactor";
+export { createInspector } from "./inspector";
+export { Page } from "./page";
+export { App } from "./app";
+export { read } from "./read";
+export { perform } from "./perform";
+export { fillIn } from "./fill-in";
+export { focused, focus, blur } from "./focused";
+export { isVisible } from "element-is-visible";
```
Comment (cowboyd) [inline packages/html/src/index.ts:17]: Most of the changes in this file seem completely irrelevant to improving click emulation. If we want to add linting or formatting tools, this should be done as a separate PR and in an automated fashion so that everybody will stay in sync.

Please review your commits before submitting to make sure that they directly pertain to the change you are proposing.

File: packages/html/testing-library.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// NOTE: Use this file as a fallback for utils those don't support `exports` field in package.json
+/** @type {import('./src/testing-library')} */
+module.exports = require("./dist/cjs/testing-library");
```
Comment (cowboyd) [inline packages/html/testing-library.js:3]: This is ok if it works. We just need to test and make sure that it doesn't change the behavior of the libraries that _do_ support package exports. In other words, we don't want it to overwrite all the cases (especially an ESM bundler).

**Review by cowboyd — COMMENTED** [review]
Except for the extra diffs related to formatting, this looks good to me.

**Review by cowboyd — APPROVED** [review]
